src/oemof/solph/flows/_investment_flow.py

```diff
@@ -69,7 +53,7 @@ def __init__(
         existing=0,
         nonconvex=False,
         offset=0,
-        **kwargs,
+        **flow_kwargs
```

Comment:
Here, you just rename kwargs to flow_kwargs but still allow arbitrary keyword arguments.

Reply:
Yes, however I added a check for the flow attributes below - so does this not count as pseudo- explicit? ;-)

Actually, I am wondering how to proceed here probably, and not repeat the whole default kwargs setting from the Flow class: [https://github.com/oemof/oemof-solph/blob/0c5031daf3ef816f9cb6f6305ea5b2dff84a24fd/src/oemof/solph/flows/_flow.py#L141-L209](https://github.com/oemof/oemof-solph/blob/0c5031daf3ef816f9cb6f6305ea5b2dff84a24fd/src/oemof/solph/flows/_flow.py#L141-L209)

In the end, it would make sense to make the kwargs of the Flow class explicit first. Then, we could adapt the flow in another PR, and remove the kwargs from the NonConvexFlow and the InvestmentFlow in following PRs. What do you think?

Reply:
I am wondering how to make the flow kwargs explicit at all as we allow additional arbitrary keywords as "emission_factor" and for additional constraints. So, we need to have **kwargs anyway?

Reply:
This is the idea of `flow_kwargs`, but I wonder why there are still stars. In my opinion it should be like this:

```python
    def __init__(
        self,
        maximum=float("+inf"),
        minimum=0,
        ep_costs=0,
        existing=0,
        nonconvex=False,
        offset=0,
        flow_kwargs)
 ```
 The new API would therefore look like this:
 ```diff
- Flow(variable_costs=5, emissions=10)
+ Flow(variable_costs=5, flow_kwargs={"emissions": 10})
```

Reply:
That sounds reasonable. Therefore, this API needs still to be implemented in v0.5 branch, or am I missing something?

[https://github.com/oemof/oemof-solph/blob/fec3f07d3d3a5e560ae09ef4b78743abc08fd27c/src/oemof/solph/constraints/integral_limit.py#L87](https://github.com/oemof/oemof-solph/blob/fec3f07d3d3a5e560ae09ef4b78743abc08fd27c/src/oemof/solph/constraints/integral_limit.py#L87)

Reply:
As you do not touch the class `Flow`, you do not break the API of `Flow`. (However, I already did by moving it to a submodule `flows`.)

Reply:
> This is the idea of `flow_kwargs`, but I wonder why there are still stars. In my opinion it should be like this:
> 
> ```python
>     def __init__(
>         self,
>         maximum=float("+inf"),
>         minimum=0,
>         ep_costs=0,
>         existing=0,
>         nonconvex=False,
>         offset=0,
>         flow_kwargs=None)
> ```
> 
> The new API would therefore look like this:
> 
> ```diff
> - Flow(variable_costs=5, emissions=10)
> + Flow(variable_costs=5, flow_kwargs={"emissions": 10})
> ```

I like the idea of having to explicitly place the additional arguments in a dictionary. This raises awareness for the user. What is more, is that typos for the standard attributes are caught, when you e.g. type `ep_cost` and expect `ep_costs` being set. This is not caught in the current implementation, though. So I suggest to include the change proposed here by @uvchik.

Reply:
Okay, how should we proceed?

In my opinion it makes sense to first implement the explicit keywords in the the Flow, and then proceed with this PR. (And then, also the NonConvexFlow)

Reply:
I'd just follow @jokochems' suggestion. It's easier to address the explicit kwargs in a second PR when this one is already merged.